Eastern-Research-Group/mywaterway
# Feature/316 tribal upstream watershed

## Related Issues:
* [HMW-316](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-316)

## Main Changes:
* Added the _Upstream Watershed_ widget to the **Tribe** page.
* Altered the widget behavior to allow the user to select a watershed to view its upstream.
* Included an instruction box for the widget.
* Changed text on the main **State & Tribal** page.

## Steps To Test:
1. Visit the Red Lake Band of Chippewa Indians page, i.e. [http://localhost:3000/tribe/REDLAKE](http://localhost:3000/tribe/REDLAKE).
2. Click the _Upstream Watershed_ widget.
3. Check that a close-able instruction box appears and that the widget icon indicates activity.
4. Zoom in until watershed boundary lines appear, and confirm that the instructions update.
5. Click inside a watershed and confirm that the upstream watershed is shown and positioned correctly.
6. Click the widget again, and confirm that the extent returns to its previous location and the upstream watershed disappears.
7. Activate the widget, then click outside the map; the widget should exit "selection mode". Alternatively, click the widget again to deactivate.
8. Activate the widget, then click an area in one of the great lakes to verify errors are handled properly.
9. Verify that the proper text has been changed on the **State & Tribal** page.
